src/subprocess_launcher.rs

```diff
@@ -0,0 +1,51 @@
+#[cfg(any(feature = "v2_40", feature = "dox"))]
+use SubprocessLauncher;
+use ffi;
+use glib;
+use glib::object::IsA;
+use GtkRawFd;
+
+pub trait SubprocessLauncherExtManual {
+    #[cfg(any(feature = "v2_40", feature = "dox"))]
+    fn take_fd(&self, source_fd: GtkRawFd, target_fd: GtkRawFd);
```

Comment:
These should all take a https://doc.rust-lang.org/nightly/std/os/unix/io/trait.IntoRawFd.html on UNIX and https://doc.rust-lang.org/nightly/std/os/windows/io/trait.IntoRawHandle.html on Windows to be safe (the safety aspect is then moved to the implementors of that trait and people can safely pass various types in there)

Reply:
See e.g. https://github.com/sdroege/gstreamer-rs/blob/master/gstreamer/src/bus_windows.rs and https://github.com/sdroege/gstreamer-rs/blob/master/gstreamer/src/bus_unix.rs

Reply:
or https://github.com/gtk-rs/gio/blob/master/src/socket.rs

Reply:
Was wrong, my link only for sockets

Reply:
Basically same thing though, yes

Reply:
Yes, I had unsafe in mind, but I don't get why I didn't think about `AsRawFd` and `AsRawHandle`...

Reply:
Into, not As :) this takes ownership. And it does not have to be unsafe then because the trait impls already take care of that being safe

Reply:
True!